README.md

```diff
@@ -1,2 +1,74 @@
+More details about how to use them are included in other documentation.
+There is no context in this repo to keep it as generic as possible.
+(i.e. it shouldn't be exclusive to any individual Matomo setup.)
+
+### download-logs
+
+This script can be used to download logs from AWS CloudWatch.
+It assumes the logs in question are inside a log group called "matomo".
+Pass it two millisecond timestamps and it will download any errored requests from Nginx between those timestamps.
+For example, to get logs from between 9:32 and 9:49 on 2019-01-01, pass in `1546507920000` and `1546508999999`:
+
+```
+cd download-logs
+./download.sh 1546507920000 1546508999999
+```
+
+The script currently prompts you to continue, advising that you specify `--log-stream-name` arguments.
+Doing so will make the request faster, since it only has to check the log streams specified.
+If you don't know the streams that need including, or you want to be sure you get all logs within that timestamp, it should be fine to continue without these, but be aware the request will take longer as a result.
+(If it's running for 10+ minutes for 150k log events across a period of a day, then something might be up.)
+
+Note that **you must be authenticated with AWS for this script to run**.
+
+The script writes its output to a file named `aws_logs.txt` in the current directory.
+
+### filter-logs
+
+This script can be used to transform the output of `download-logs` back into an Nginx JSON format.
+It essentially strips the wrapper from the response to `download-logs` above, and then transforms the JSON array of events (where each element spans multiple lines) into a newline-delimited JSON (ndjson: each event is a separate JSON entity, on a single line) that the Matomo log analytics script can ingest.
+It is a Python 3 script, and expects to be passed as an argument a single file to transform, for example:
+
+```
+python3 filter-logs/filter-cloudwatch-logs.py download-logs/aws_logs.txt
+```
+
+By default, the above command would output a file `download-logs/aws_logs.txt.ndjson`.
+Note that if that output file already exists, the transformed output will be appended, rather than overwrite the existing data.
+
+### replay-events
+
+This is essentially a Dockerfile used to define an environment with Python 2 to run the Matomo log analytics script.
+A fork is being used to handle `X-Forwarded-For` headers as given by AWS Application ELBs.
+The command itself has to be run manually once inside the docker image, as token authentication using Matomo hasn't worked in testing thus far, so authentication credentials need to be supplied directly.
+An example workflow might be something like this:
+
+```
+mv download-logs/aws_logs.txt.ndjson replay-events/access.log
+cd replay-events
+docker build -t matomo-replay-events .
+docker run --rm -it matomo-replay-events
+# inside docker container:
+python -u /log-analytics/import_logs.py \
```

Comment:
Any reason you wouldn't make this command the entry point for Docker image and bind mount the access.log file at runtime?

Reply:
I didn't want to make it the entrypoint because of needing the credentials. You'd have to override all the arguments to specify the correct credentials, so I thought it was cleaner to just drop into a shell if you run the container without a specific override. Is there a better way of doing all this? (Short of getting the PHP auth to work... But the error messages I was getting were not particularly useful...)

As for the bind mount, you're right, that would be better. Especially for a large log file. I'll update. 👍

Reply:
Ahh, thanks. That's helpful. I didn't know about the `-e` flag. I'll make some changes based on this and update the Readme. I guess I should add checks for non-empty values for those environment variables into the bash script too.

Reply:
Right, I think I've found a combination that works. Thanks for all your ideas and help. Any further improvements that can be made, do suggest!